<!doctype html>
<html lang="en">

  <head>
    <meta charset="utf-8">

    <title>Nanopore Sequencing</title>

    <meta name="viewport" content="width=device-width, initial-scale=1" />

    <link rel="manifest" href="manifest.json" />
    <link rel="stylesheet" href="../lib/reveal-4.4.0/reveal.css">
    <link rel="stylesheet" href="../lib/reveal-themes/sky.css" id="theme">
    <link rel="stylesheet" href="../lib/highlight-themes/base16/solarized-light.css" id="highlight-theme">
    <link rel="stylesheet" href="../lib/font-awesome/css/all.css">
    <style>
      .reveal p {
        text-align: left;
      }
      .reveal ul {
        display: block;
      }
      .figure-right {
        float: right;
        margin: 0.5em 0 0 0.5em;
      }
      .figure-right > img {
        margin: 0;
      }
      .reveal.overview .slides > section > section {
        padding-top: 2em;
      }
      .reveal, .reveal h1, .reveal h2, .reveal h3, .reveal h4, .reveal h5, .reveal h6 {
        font-family: Palatino Linotype, Book Antiqua, Palatino, FreeSerif, serif;
        letter-spacing: normal;
        text-transform: none;
      }
    </style>
  </head>

  <body>

    <div class="reveal">

      <div class="slides">

        <section>
          <h2>Nanopore DNA Sequencing</h2>
          <h4>Data Science Engineering for Single-Molecule Sensors</h4>
          <section data-markdown>
            <textarea data-template>
              <!-- .slide: data-menu-title="Introduction" -->

              <br>

              ###### Jesse Clark

              <br>

              ###### 2023

              Notes:
              Hello. I'm a software engineer at Roche Diagnostics.
              I'd like to tell you about the work I've been doing on our nanopore sequencing instrument.
              I'm familiar with the data science operations for this project, and how they interact with the electrical and chemical systems that we use, and the markets that we operate in.

              First, I will give some background about DNA sequencing technologies.

              Then, I will explain the algorithms involved in primary analysis of nanopore data.

              ---

              #### **Abstract**

              DNA sequencing has a long history of tradeoffs between cost and accuracy, mediated by data science.

              As improvements in biochemistry deliver raw data with better signal-to-noise ratio, the role of conventional bioinformatics in primary analysis is being supplanted by recurrent neural networks.

              A case study of the record-breaking high-throughput process at Roche Sequencing Solutions.
            </textarea>
          </section>
        </section>

        <section>
          <h3>Conventional Sequencing</h3>
          <section data-markdown>
            <textarea data-template>
              <!-- .slide: data-menu-title="Conventional Sequencing" -->

              <figure class="figure-right" style="width:50%">
                <img src="2021_Sequencing_cost_per_Human_Genome.jpg"/>
              </figure>

              [The Human Genome Project](https://www.genome.gov/human-genome-project)
              first sequenced the human genome  
              (3.1 billion base pairs) in 2003.

              Since then, we have seen three distinct generations of sequencing technology, characterized by higher signal-to-noise ratio and lower costs.

              Increasing cost-effectiveness opens up new markets:

              - basic research in genetics
              - non-invasive prenatal testing
              - early cancer detection
              - large-scale infectious disease research

              Notes:
              In each generation, we see incremental improvements due to massively parallel instruments, streamlined processes such as "click chemistry".

              Breakthroughs are due to new techniques.

              ---

              ### Sequencing by Ligation

              <figure class="figure-right" style="width:45%">
                <img src="Gel_Electrophoresis.svg" />
              </figure>

              - Cut DNA strands into fragments  
              - Attach base-dependent fluorescence to the end of each fragment  
              ("chain termination method")
              - Separate fragments by size  
              ([Gel Electrophoresis](https://en.wikipedia.org/wiki/Gel_electrophoresis))

              Signal gets weaker at longer read lengths. The sample can be amplified with [PCR](https://en.wikipedia.org/wiki/Polymerase_chain_reaction), but that introduces noise dependent on earlier noise.

              High [depth of coverage](https://en.wikipedia.org/wiki/Coverage_(genetics)) is needed to differentiate between mutations in the sample and mutations in the process.

              Notes:
              Electrophoresis sequencing (c. 1977)

              PCR (c. 1985)

              Data scientists can correlate non-synchronized data *in silico* during primary analysis.

              Roche is doing interesting work with neural network early interpretation of PCR curves. Faster turnaround time can open new markets.

              ---

              ### Sequencing by Synthesis

              <br>

              Synchronized reaction and detection.

              For example, engineered nucleotides that fluoresce when added to a chain.

              Optical detection of fluorescence during DNA synthesis.
            </textarea>
          </section>
        </section>

        <section>
          <h3>Nanopore Sequencing</h3>
          <section data-markdown>
            <textarea data-template>
              <!-- .slide: data-menu-title="Nanopore Sequencing" -->

              <figure class="figure-right" style="width:50%">
                <img src="Single_channel.png">
                <figcaption><small><i>Voltage across a single ion channel</i></small></figcaption>
              </figure>

              A [patch clamp](https://en.wikipedia.org/wiki/Patch_clamp) can measure voltage or current flow through a single ion&nbsp;channel.

              [Nanopore sequencing](https://en.wikipedia.org/wiki/Nanopore_sequencing) is based on threading a single strand of DNA through an ion channel and measuring the characteristic voltage signature of each nucleotide.

              This technique is sensitive enough to measure single molecules of native DNA without introducing noise before or after the main reaction.

              Notes:
              Patch clamp (early 1980s)

              Nanopore: a pore about 1 nanometer in diameter.

              ---
  
              ### Ion Channel Threading <!-- .element: style="display:none" -->

              <figure class="figure-right">
                <img src="oxford-nanopore-2009.jpg" style="height: 10em" />
                <figcaption><small><i>Oxford Nanopore (2009)</i></small></figcaption>
              </figure>

              - Create a membrane with a single pore.  
              Pore-forming toxins such as Î±-hemolysin can be produced by bacteria.

              - Attach a DNA-threading enzyme to the pore.

              - Measure voltage across the pore as the DNA passes through it.

              <br style="clear:right">

              Biochemical engineers can modify nucleotides to make their electric signatures more distinct ("sequencing by synthesis").

              ---

              <!-- .slide: data-menu-title="High-Throughput Platform" -->

              ### Roche High-Throughput Platform

              <figure class="figure-right">
                <img src="roche-cobas-6800-product-shot.png" style="width:18em" />
              </figure>

              - Custom chip with 8,000,000<br>single-pore wells and a general purpose rolling-shutter array of patch&nbsp;clamp&nbsp;sensors  
              (Genia Technologies)

              - Engineered nucleotides with a ratcheting effect that regulates procession rate  
              (Stratos Genomics)

              Notes:
              SBX makes it much easier to detect and prevent insertions and deletions, giving us a very high accuracy with fewer samples per base dwell.

              Genia (2009-2014)

              Stratos (2007-2020)
            </textarea>
          </section>
        </section>

        <section>
          <h3>Primary Data Analysis</h3>
          <section data-markdown>
            <textarea data-template>
              <!-- .slide: data-menu-title="Primary Data Analysis" -->

              Primary analysis of sequencing data produces:
              - sequence of DNA base-calls
              - quality-control metrics
              - genome alignment (if applicable)

              These can then be passed into later analysis stages such as mutation calling.

              <br>

              **Algorithms need to be adapted to the specific chemistry they are working with.**

              Notes:
              The cost of sequencing is usually dominated by hardware and reagents.

              The analysis algorithms are straightforward, but they need to be adapted to the specific chemistry they are working with.

              A well-run data science team should be able to make changes quickly and enable experimentalists to self-manage code versions and parameters.

              ---

              ### Pipeline

              <!-- <img src="Primary Data Analysis.svg" style="width:90%" /> -->
              <object type="image/svg+xml" data="Primary Data Analysis.svg" style="width:90%" />

              Notes:
              Here is how we handle primary data analysis at Roche.

              - Experiment metadata (chemical reagents, electric waveforms, relevant genome, ...) is specified in advance. Voltage data is collected as "frames" which consist of millions of 8-bit voltage readings, plus some "frame header" metadata.

              - Data is transposed from frame-major order into cell-major order, so that the complete record of a single cell is available in a single data chunk. Cells are subsampled randomly into chunks. These chunks are copied to cloud storage and tracked with an in-house database.

              - Data in the cloud is analyzed on-demand by a Docker image (with versions, branches, parameters, ...). Data chunks are analyzed in parallel and then the results are merged. Results include essential base-call information as well as extra information for algorithm development and tuning.

              - Data in the cloud can be viewed with an in-house data visualization tool.

              - Analysis results are used to train a neural network to perform base-calls in real time on the production instrument.

              ---

              ### Algorithms

              - Pore lifetime detection (including various dysfunctional states)

              - Voltage normalization (open-channel voltage drifts over time)

              - Molecular trace detection

              - Level-calling

              - Base-calling

              - Genome alignment

              - Quality metrics (read length statistics, confidence of each base-call, ...)

              - Neural network training
            </textarea>
          </section>
        </section>
        <section data-markdown>
          <textarea data-template>
            <!-- .slide: data-menu-title="Closing" -->

            <img src="Roche-logo.svg" class="plain" height="75"/>

            #### Doing now what patients need next

            <br><br>

            ##### Jesse Clark
            ###### `jesse@modusponens.org`
          </textarea>
        </section>
      </div>
    </div>

    <script src="../lib/reveal-4.4.0/reveal.js"></script>
    <script src="../lib/reveal-plugins/markdown/markdown.js"></script>
    <script src="../lib/reveal-plugins/highlight/highlight.js"></script>
    <script src="../lib/reveal-plugins/notes/notes.js"></script>
    <script src="../lib/reveal-plugins/math/math.js"></script>
    <script src="../lib/reveal-plugins/search/search.js"></script>
    <script src="../lib/reveal-plugins/menu/menu.js"></script>
    <script src="../lib/reveal-plugins/zoom/zoom.js"></script>

    <script type="text/template" id="menu-tools">
      <ul class="slide-menu-items">
        <li class="slide-menu-item active">
          <div onclick="document.documentElement.requestFullscreen(document.documentElement)">
            <i class="fas fa-expand active"></i>
            <span >Fullscreen</span>
          </div>
        </li>
        <li class="slide-menu-item active">
          <div onclick="Reveal.toggleOverview()">
            <i class="fas fa-th-large active"></i>
            <span >Overview</span>
          </div>
        </li>
      </ul>
    </script>

    <script>

      Reveal.initialize({
        width: 1440,
        height: 900,
        margin: 0.05,
        totalTime: 60 * 20,
        center: false,
        disableLayout: false,
        controls: (document.location.protocol != 'file:'), //(navigator.maxTouchPoints >= 1),
        controlsTutorial: false,
        controlsLayout: (navigator.maxTouchPoints >= 1 ? 'edges': 'bottomRight'),
        progress: false,
        slideNumber: "c/t",
        hashOneBasedIndex: true,
        showNotes: false,
        hash: true,
        history: true,
        center: true,
        navigationMode: "linear",
        transition: "convex",
        transitionSpeed: "slow",
        pdfSeparateFragments: false,
        markdown: {
          smartypants: true
        },
        katex: {
          local: '../lib/katex-0.16.4'
        },
        menu: {
          sticky: false,
          numbers: false,
          transitions: true,
          loadIcons: false,
          themesPath: '../lib/reveal-themes/',
          custom: [
            {
              title: 'Tools',
              icon: '<i class="fa fa-toolbox">',
              content: document.getElementById("menu-tools").innerHTML
            }        
          ],
          themes: [
            { name: 'Beige', theme: '../lib/reveal-themes/beige.css',
              highlightTheme: '../lib/highlight-themes/github.css' },
            { name: 'Black', theme: '../lib/reveal-themes/black.css',
              highlightTheme: '../lib/highlight-themes/monokai.css' },
            { name: 'Blood', theme: '../lib/reveal-themes/blood.css',
              highlightTheme: '../lib/highlight-themes/monokai-sublime.css' },
            { name: 'League', theme: '../lib/reveal-themes/league.css',
              highlightTheme: '../lib/highlight-themes/github-dark-dimmed.css' },
            { name: 'Moon', theme: '../lib/reveal-themes/moon.css',
              highlightTheme: '../lib/highlight-themes/monokai.css' },
            { name: 'Night', theme: '../lib/reveal-themes/night.css',
              highlightTheme: '../lib/highlight-themes/github-dark.css' },
            { name: 'Serif', theme: '../lib/reveal-themes/serif.css',
              highlightTheme: '../lib/highlight-themes/base16/solarized-light.css' },
            { name: 'Simple', theme: '../lib/reveal-themes/simple.css',
              highlightTheme: '../lib/highlight-themes/github.css' },
            { name: 'Sky', theme: '../lib/reveal-themes/sky.css',
              highlightTheme: '../lib/highlight-themes/github.css' },
            { name: 'Solarized', theme: '../lib/reveal-themes/solarized.css',
              highlightTheme: '../lib/highlight-themes/base16/solarized-light.css' },
            { name: 'White', theme: '../lib/reveal-themes/white.css',
              highlightTheme: '../lib/highlight-themes/github.css' }
          ]
        },
        plugins: [
          RevealMarkdown,
          RevealHighlight,
          RevealNotes,
          RevealMath.KaTeX,
          RevealSearch,
          RevealMenu,
          RevealZoom
        ]
      });
    </script>

  </body>
</html>
